framework/src/application/node/synchronizer/synchronizer.js

```diff
@@ -141,6 +153,93 @@ class Synchronizer {
 
 		return undefined;
 	}
+
+	async loadUnconfirmedTransactions() {
+		for (const index of new Array(this.loadTransactionsRetries).keys()) {
```

Comment:
```
while (this.loadTransactionsRetries >= 0) {
this.loadTransactionsRetries --;
  try {
				await this._getUnconfirmedTransactionsFromNetwork();

				break;
			} catch (err) {
				if (err && index === this.loadTransactionsRetries - 1) {
					this.logger.error(
						{ err },
						`Failed to get transactions from network after ${this.loadTransactionsRetries} retries`,
					);
				}
			}
}
```

Reply:
I think simple `for (let retry = 0; retry < this. loadTransactionsRetries; retry ++)` is a lot simpler